.github/workflows/build-container.yaml

```diff
@@ -28,16 +23,16 @@ on:
 env:
   REGISTRY: ghcr.io
   IMAGE_NAME: ${{ github.repository }}
-  TAG: ghcr.io/${{ github.repository }}:${{ github.event_name == 'pull_request' && format('pr-{0}', fromJSON(github.event.pull_request.number)) || (inputs.tag || 'latest') }}
-  PUBLISH: ${{ github.event_name == 'release' || (github.event.workflow_dispatch.ref == '/refs/head/main' && inputs.publish) }}
+  PUBLISH: ${{ github.event_name == 'push' && startsWith(github.event.push.ref, 'refs/tag') || (github.event.workflow_dispatch.ref == '/refs/head/main' && inputs.publish) }}
```

Comment:
TAG is auto generated and publishing is predicated on a tag or a workflow dispatch from main.